client/mainmenu/CMainMenu.cpp

```diff
@@ -532,16 +540,29 @@ void CMultiMode::onNameChange(std::string newText)
 	name->String() = newText;
 }
 
-CMultiPlayers::CMultiPlayers(const std::vector<std::string> & playerNames, ESelectionScreen ScreenType, bool Host, ELoadMode LoadMode)
+CMultiPlayers::CMultiPlayers(const std::vector<std::string>& playerNames, ESelectionScreen ScreenType, bool Host, ELoadMode LoadMode, EShortcut shortcut)
 	: loadMode(LoadMode), screenType(ScreenType), host(Host)
 {
 	OBJECT_CONSTRUCTION;
 	background = std::make_shared<CPicture>(ImagePath::builtin("MUHOTSEA.bmp"));
 	pos = background->center(); //center, window has size of bg graphic
 
-	std::string text = CGI->generaltexth->allTexts[446];
-	boost::replace_all(text, "\t", "\n");
-	textTitle = std::make_shared<CTextBox>(text, Rect(25, 10, 315, 60), 0, FONT_BIG, ETextAlignment::CENTER, Colors::WHITE); //HOTSEAT	Please enter names
+	std::string text;
+	switch (shortcut)
+	{
+	case EShortcut::MAIN_MENU_HOTSEAT:
+		text = CGI->generaltexth->allTexts[446];
+		boost::replace_all(text, "\t", "\n");
+		break;
+	case EShortcut::MAIN_MENU_HOST_GAME:
+		text = CGI->generaltexth->translate("vcmi.mainMenu.hostTCP");
+		break;
+	case EShortcut::MAIN_MENU_JOIN_GAME:
+		text = CGI->generaltexth->translate("vcmi.mainMenu.joinTCP");
+		break;
+	}
```

Comment:
Shouldn't existing `ESelectionScreen` or `ELoadMode` enums suffice?

Reply:
They are not enough as we need "sub multiplayer entry". Have not found better way to get it, only from EShortcut.

```
ESelectionScreen = newGame, loadGame, saveGame, scenarioInfo, campaignList
ELoadMode = NONE, SINGLE, MULTI, CAMPAIGN, TUTORIAL
```